apps/wappush/js/messagedb.js

```diff
@@ -76,18 +102,27 @@ var MessageDB = (function() {
 
       req.onsuccess = function mdb_opened(event) {
         db = event.target.result;
-        resolve(db);
+        mdb_deleteViewedMessages().then(() => resolve(db));
       };
       req.onerror = function mdb_openError(event) {
         reject(event.target.error);
       };
       req.onupgradeneeded = function mdb_upgradeNeeded(event) {
         var db = event.target.result;
-
-        if (event.oldVersion === 0) {
-          var store = db.createObjectStore(MESSAGE_STORE_NAME,
-                                           { keyPath: KEY_PATH });
-          store.createIndex(ID_INDEX, ID_INDEX_KEYPATH, { unique: false });
+        var store = null;
+
+        switch (event.oldVersion) {
+          case 0:
+            store = db.createObjectStore(MESSAGE_STORE_NAME,
+                                         { keyPath: KEY_PATH });
```

Comment:
Good point, it's awkward for re-factoring and if you have long function names. And yes I did some gecko work lately ;-)